BusBooking.cls

```diff
@@ -0,0 +1,140 @@
+    capacitylist = new List<SelectOption>();
+    capacitylist.add(new SelectOption('0','0'));
+    capacitylist.add(new SelectOption('1','1'));
+    capacitylist.add(new SelectOption('2','2'));
+    capacitylist.add(new SelectOption('3','3'));
+    capacitylist.add(new SelectOption('4','4'));
+    capacitylist.add(new SelectOption('5','5'));
+    capacitylist.add(new SelectOption('6','6'));
+    capacitylist.add(new SelectOption('7','7'));
+    capacitylist.add(new SelectOption('8','8'));
+    capacitylist.add(new SelectOption('9','9'));
+    capacitylist.add(new SelectOption('10','10'));
+    return capacitylist;
+   }
+   
+   
+   /** List to display the schoolstops**/
+    public List<SelectOption> getStopList() {
+     stops= new List<SelectOption>();
+     stops.add(new SelectOption('None','None'));
+     stops.add(new SelectOption('Symbiosis School', 'Symbiosis School')); 
+     stops.add(new SelectOption('DAV School', 'DAV School'));
+     stops.add(new SelectOption('Army Public School', 'Army Public School')) ;  
+     return stops;  
+      } 
+      
+     
+    /** Get routes method to display the routes based on google api**/    
+     public void getroutes(){
+       HttpRequest req = new HttpRequest();
+       Http http = new Http();
+       req.setMethod('GET');
+       String url='https://maps.googleapis.com/maps/api/geocode/json?key=AIzaSyA0vqKj6rA67xgzWviPdr6GknQsy-WUWmM&address='+acc.BillingPostalCode+'&sensor=false';  
+       req.setEndPoint(url);
+       HTTPResponse resp = http.send(req);       
+       JSONParser parser = JSON.createParser(resp.getBody()); //Json parser to parse the json string obtained from url
+       Double latitude = null;
+       Double longitude = null;
+       while (parser.nextToken() != null)
+       {
+       if ((parser.getCurrentToken() == JSONToken.FIELD_NAME)  && (parser.getText() == 'location'))
+       {
+       // Jump to latitude and longtitude fields.
+       parser.nextToken();
+       while (parser.nextToken() != JSONToken.END_OBJECT)
+       {
+       String text = parser.getText();
+       parser.nextToken();
+       if (text == 'lat') latitude = parser.getDoubleValue();
+       else if (text == 'lng') longitude = parser.getDoubleValue();
+          }
+          } 
+          }
+       String query = 'SELECT School__c FROM Stop__c WHERE';
+       query+= ' DISTANCE(Location__c, GEOLOCATION('+latitude+','+longitude+'),\'mi\') < 10 ';
+       query+='ORDER BY DISTANCE(Location__c, GEOLOCATION('+latitude+','+longitude+'),\'mi\') LIMIT 3';       
+       if(latitude != null && longitude!= null){
+       List<Stop__c> stoplist = Database.query(query);            
+       Set<id> setstopid = new Set<Id>();             
+       for(Stop__C obj:stoplist){
+       setstopid.add(obj.id);
+          }             
+       lstroutestop=[SELECT Route__c,Route__r.name,Stop__r.name,Route__r.From__c,Route__r.To__c FROM Route_Stop__c where Stop__c in :setstopid ];
```

Comment:
Use below query instead of removing it manually from list.
`lstroutestop=[SELECT Route__c,Route__r.name,Stop__r.name,Route__r.From__c,Route__r.To__c FROM Route_Stop__c where Stop__c in :setstopid and  Stop__r.name =: selectedstops];`